script.js

```diff
@@ -1,4 +1,493 @@
+      ", your bet is " +
+      playerProfiles[currPlayer].playerCurrRoundBet;
+
+    if (currPlayer + 1 < numPlayers) {
+      gameMode = GAME_MODE_BETTING_MODE;
+
+      currPlayer += 1;
+      playerBetMessage =
+        playerBetMessage +
+        ". <br><br> Player " +
+        playerProfiles[currPlayer].id +
+        " , how much would you like to bet?";
+      return playerBetMessage;
+    } else {
+      gameMode = GAME_MODE_PLAYERS_TURN;
+      var allBetsInmessage = "All bets in! Continue to draw cards, good luck!";
+      return allBetsInmessage;
+    }
+  }
+}
+function drawingPlayersCardsAndDealerFirstCard(deck) {
+  for (var j = 0; j < 2; j++) {
+    for (currPlayer = 0; currPlayer < playerProfiles.length; currPlayer++) {
+      card = deck.pop();
+      playerProfiles[currPlayer].playerHand +=
+        "<br>" + card.value + " of " + card.suits;
+      playerProfiles[currPlayer].playerHandValue += getValue(card);
+      playerProfiles[currPlayer].playerAceCount += checkAce(card);
+      while (
+        playerProfiles[currPlayer].playerHandValue > 21 &&
+        playerProfiles[currPlayer].playerAceCount > 0
+      ) {
+        playerProfiles[currPlayer].playerHandValue -= 10;
+        playerProfiles[currPlayer].playerAceCount -= 1;
+      }
+    }
+  }
+
+  gameMode = GAME_MODE_HIT_OR_STAND;
+
+  for (var j = 0; j < playerProfiles.length; j++) {
+    if (playerProfiles[j].playerHandValue == 21) {
+      playerProfiles[j].playerRoundScore += "BLACKJACK! - ";
+    }
+  }
+
+  let dealerCard = deck.pop();
+  dealerProfile.dealerHand +=
+    "<br>" + dealerCard.value + " of " + dealerCard.suits;
+  dealerProfile.dealerHandValue += getValue(dealerCard);
+  dealerProfile.dealerAceCount += checkAce(dealerCard);
+
+  var playerHandmessage = createPlayerHandmessage();
+  return (
+    playerHandmessage +
+    "<br><br> Player " +
+    playerProfiles[0].id +
+    ", hit or stand?"
+  );
+}
+function getValue(card) {
+  let value = card["value"];
+
+  if (isNaN(value)) {
+    if (value == "Ace") {
+      return 11;
+    }
+    return 10;
+  }
+  return parseInt(value);
+}
+
+function checkAce(card) {
+  if (card["value"] == "Ace") {
+    return 1;
+  }
+  return 0;
+}
+function createPlayerHandmessage() {
+  var playerHandmessage =
+    "Dealer card: " +
+    dealerProfile.dealerHand +
+    "<br> Dealer current points: " +
+    dealerProfile.dealerHandValue +
+    "<br>";
+
+  for (i = 0; i < playerProfiles.length; i++) {
+    var playerHandmessage =
+      playerHandmessage +
+      "<br><br>Player " +
+      playerProfiles[i].id +
+      "'s cards:" +
+      playerProfiles[i].playerHand +
+      "<br> Current points: " +
+      playerProfiles[i].playerHandValue +
+      "<br> Chips balance: " +
+      playerProfiles[i].playerChipsBalance +
+      "<br> Bet this round: " +
+      playerProfiles[i].playerCurrRoundBet;
+  }
+  return playerHandmessage;
+}
+function toValidate_HitorStand(input) {
+  var validation = input.toUpperCase();
```

Comment:
This function is getting alittle too long. You could create some helper functions to handle when the user has an Ace etc...